test/integration/testpaths/testpath_netmask.py

```diff
@@ -0,0 +1,152 @@
+            cls.zone.id,
+            cls.testdata["ostype"])
+
+        cls._cleanup = []
+
+        cls.skiptest = False
+
+        if cls.hypervisor.lower() not in ["xenserver"]:
+            cls.skiptest = True
+            return
+
+        try:
+
+            # Create an account
+            cls.account = Account.create(
+                cls.apiclient,
+                cls.testdata["account"],
+                domainid=cls.domain.id
+            )
+            cls._cleanup.append(cls.account)
+
+            # Create user api client of the account
+            cls.userapiclient = testClient.getUserApiClient(
+                UserName=cls.account.name,
+                DomainName=cls.account.domain
+            )
+
+            # Create Service offering
+            cls.service_offering = ServiceOffering.create(
+                cls.apiclient,
+                cls.testdata["service_offering"],
+            )
+            cls._cleanup.append(cls.service_offering)
+
+            cls.vm = VirtualMachine.create(
+                cls.apiclient,
+                cls.testdata["small"],
+                templateid=cls.template.id,
+                accountid=cls.account.name,
+                domainid=cls.account.domainid,
+                serviceofferingid=cls.service_offering.id,
+                zoneid=cls.zone.id,
+                mode=cls.zone.networktype
+            )
+
+        except Exception as e:
+            cls.tearDownClass()
+            raise e
+        return
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            cleanup_resources(cls.apiclient, cls._cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+
+    def setUp(self):
+        if self.skiptest:
+            self.skipTest(
+                "Test not to be run on %s" %
+                self.hypervisor)
+        self.apiclient = self.testClient.getApiClient()
+        self.dbclient = self.testClient.getDbConnection()
+        self.cleanup = []
+
+    def tearDown(self):
+        try:
+            cleanup_resources(self.apiclient, self.cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+        return
+
+    @attr(tags=["basic"], required_hardware="false")
+    def test_01_netmask_value_check(self):
+        """ Check Netmask value in database
+            1. Check if netmask attribute in nics table
+                stores correct value.
+        """
+
+        qryresult_netmask = self.dbclient.execute(
+            " select id, uuid, netmask\
+                    from nics where netmask = '%s';" %
+            self.vm.nic[0].netmask)
+
+        self.assertNotEqual(
+            len(qryresult_netmask),
+            0,
+            "Check if netmask attribute in nics table \
+            stores correct value")
+
+        netmask_id = qryresult_netmask[0][2]
+        if netmask_id != "NULL":
+            netmask_value = re.match(
```

Comment:
This regex only checks for valid netmask. However, it does not validate whether the netmask is from the guest cidr.